src/vs/base/browser/ui/iconLabel/iconLabel.ts

```diff
@@ -259,7 +259,7 @@ export class IconLabel extends Disposable {
 						};
 						IconLabel.adjustXAndShowCustomHover(hoverOptions, mouseX, hoverDelegate, isHovering);
 
-						const resolvedTooltip = await tooltip(tokenSource.token);
+						const resolvedTooltip = (await tooltip(tokenSource.token)) ?? (!isString(markdownTooltip) ? markdownTooltip.markdownNotSupportedFallback : undefined);
 						if (resolvedTooltip) {
```

Comment:
Is it intentional that it will not work when the empty string is returned? Wouldn't it then still show "Loading..."?

Reply:
Good catch, yes it will still show "Loading...". I think in this case would should simply hide the hover since none was provided.